src/nbl/system/ISystem.cpp

```diff
@@ -1,15 +1,14 @@
 #include "nbl/system/ISystem.h"
 #include "nbl/system/ISystemFile.h"
 #include "nbl/system/CFileView.h"
+#include "nbl/builtin/CArchive.h"
 
 #include "nbl/system/CArchiveLoaderZip.h"
 #include "nbl/system/CArchiveLoaderTar.h"
 
-
 using namespace nbl;
 using namespace nbl::system;
 
-
```

Comment:
I'm still waiting for you to nuke `impl_loadEmbeddedBuiltinData`

Reply:
also nuke from the header
```cpp
        template<nbl::core::StringLiteral Path>
        inline core::smart_refctd_ptr<const IFile> loadBuiltinData()
```

Reply:
also remove this block from `ISystem::createFile`
```cpp
    // try builtins
    if (!(flags.value&IFile::ECF_WRITE) && builtin::hasPathPrefix(filename.string()))
    {
        #ifdef _NBL_EMBED_BUILTIN_RESOURCES_
            auto file = impl_loadEmbeddedBuiltinData(filename.string(),nbl::builtin::get_resource_runtime(filename.string()));
            if (file)
            {
                future.set_result(const_cast<IFile*>(file.get()));
                return;
            }
        #else
            createFile(future,builtin::getBuiltinResourcesDirectoryPath()/filename,flags);
            return;
        #endif
    }
``` 


but for this to work we need a `CMountDirectoryArchive`, basically something that allows us to alias `NBL_BUILTIN_RESOURCES_DIRECTORY_PATH` to `nbl/builtin` like so in the constructor of `ISystem`:
```cpp
#ifdef _NBL_EMBED_BUILTIN_RESOURCES_
mount(core::make_smart_refctd_ptr<nbl::builtin::CArchive>(m_logger));
#else
mount(core::make_smart_refctd_ptr<nbl::system::CMountDirectoryArchive>(NBL_BUILTIN_RESOURCES_DIRECTORY_PATH,m_logger),"nbl/builtin");
#endif
```

Reply:
with this we can also remove the following block
```cpp
    if (!writeUsage && builtin::hasPathPrefix(filename.string()))
    {
        #ifdef _NBL_EMBED_BUILTIN_RESOURCES_
            std::pair<const uint8_t*, size_t> found = nbl::builtin::get_resource_runtime(filename.string());
            if (found.first && found.second)
                return true;
        #else
            if (exists(builtin::getBuiltinResourcesDirectoryPath()/filename,flags))
                return true;
        #endif
    }
```
from `ISystem::exists`

because two archives mounted at startup will take care of it transparently

Reply:
Also this
```cpp
    // first check if its a builtin path
    if (builtin::hasPathPrefix(p.string()))
        return true;
```
in `ISystem::isPathReadOnly`

Reply:
resolved